src/solidity-frontend/solidity_convert.cpp

```diff
@@ -5284,6 +5383,49 @@ bool solidity_convertert::get_expr(
     // * we had ruled out all the special cases
     // * we now confirm it is called by aother contract inside current contract
     // * func() ==> current_func_this.func(&current_func_this);
+
+    // * check if the function call has named arguments
+    // e.g. func({a: 1, b: 2});
+    // reorder the arguments based on the parameter order
+    auto it = expr.find("names");
```

Comment:
Can you move it to a dedicated function e.g. reorder_arguments which returns `clean_expr`